addons/hr/models/hr_employee.py

```diff
@@ -228,6 +231,14 @@ def _compute_avatar(self, avatar_field, image_field):
             employee[avatar_field] = avatar
         super(HrEmployee, employee_wo_user_and_image)._compute_avatar(avatar_field, image_field)
 
+    @api.depends('birthday_public_display')
+    def _compute_birthday_public_display_string(self):
+        for employee in self:
+            if employee.birthday and employee.birthday_public_display:
+                employee.birthday_public_display_string = datetime.strftime(employee.birthday, "%d %B")
```

Comment:
Please use ``format_date`` from ``odoo.tools`` to format the date according to the user's lang.

You'll need to add an ``@api.depends_context`` on user's lang as well.